syntax/BaseSyntaxPlugin.php

```diff
@@ -19,6 +19,10 @@ abstract class BaseSyntaxPlugin extends \dokuwiki\Extension\SyntaxPlugin
     protected $ptype;
     /** @var int */
     protected $sort;
+    /** @var string */
+    protected $tag;
+    /** @var string */
+    protected $mode;
```

Comment:
Here I would have the idea to have a function instead of variable `$mode`:
```php
    public function getMode(): string
    {
        return 'plugin_' . $this->getPluginName() . '_' . $this->getPluginComponent();
    }
```
Then no `$mode` would be necessary in the modules.